develop/tutorials/articles/03-developing-a-web-application/04-writing-web-application-liferay/01-writing-a-web-application.markdown

```diff
@@ -0,0 +1,635 @@
+# Writing Your First Liferay Application
+
+It's easy to get started with your first Liferay application. We'll guide you
+through the step-by-step process of creating your project and deploying your
+application to Liferay. Before you know it, you'll have your application
+deployed alongside of the applications that come with Liferay.
+
+Your first application is simple: you'll build a guestbook application that
+looks like this:
+
+![Figure x: You'll create this simple application.](../../images/first-guestbook-portlet.png)
+
+By default, it shows guestbook messages that various users have left on your
+website. To add a message, you click the *Add Entry* button to show a form that
+lets you enter and save a message.
+
+Ready to write your first Liferay application?
+
+## Creating Your First Liferay Application
+
+Let's jump right in and create your first project. Applications in Liferay
+Portal are called *portlets* so you'll create a portlet project. You'll use the
+[Blade Tools](https://dev.liferay.com/develop/tutorials/-/knowledge_base/7-0/introduction-to-blade-tools)
+command line tool to create your application. The new Liferay application
+wizards available in Liferay IDE and Liferay Developer Studio use Blade Tools
+under the hood. Thus, new projects can be created the same way regardless of
+your specific development environment or IDE.
+
+Blade Tools allows you to create a Liferay workspace. A Liferay workspace is a
+generated environment that contains both your Liferay projects and a Liferay
+bundle on which to test them. The recommended way to develop applications for
+Liferay 7 is to use a Liferay workspace. To create a Liferay workspace, follow
+the instructions in the Creating a Liferay Workspace article:
+[https://dev.liferay.com/develop/tutorials/-/knowledge_base/7-0/creating-a-liferay-workspace](https://dev.liferay.com/develop/tutorials/-/knowledge_base/7-0/creating-a-liferay-workspace)
+Once your Liferay workspace is created, create a
+`beginning-liferay-development` folder in your Liferay workspace's `modules`
+folder. Navigate to the new folder and then use this command to create your
+Guestbook application:
+
+    blade create -c GuestbookPortlet -p com.liferay.docs.guestbook com.liferay.docs.guestbook
+
+This command creates a project named `com.liferay.docs.guestbook`. `-c
+GuestbookPortlet` specifies the name of the class to be generated. `-p
+com.liferay.docs.guestbook` specifies the package in which the class is to be
+generated. By default, the `mvcportlet` template is used. You can explicitly
+specify this using the `-t mvcportlet` option. When you use the `mvcportlet`
+template, Blade Tools creates a portlet class (the name of which can be
+specified with the `-c` option) in a package (the name of which can be
+specified by the `-p` option) in your project's `src/main/java` folder. Blade
+Tools also creates your application's JSP files and language properties file in
+your project's `src/main/resources` folder.
+
```

Comment:
Rather than a paragraph, it might be better to present the options as definitions: 

The options do these things: 

`-c`: Specifies the name of the class to be generated. 

`-p com.liferay.docs.guestbook`: Specifies the package where the class is generated. 

(etc)